docs/admin/encryption.md

```diff
@@ -0,0 +1,129 @@
+# Database Encryption
+
+By default, Coder stores external user tokens in plaintext in the database.
+Database Encryption allows Coder administrators to encrypt these tokens at-rest,
+preventing attackers with database access from using them to impersonate users.
+
+## How it works
+
+Coder allows administrators to specify up to two
+[external token encryption keys](../cli/server.md#external-token-encryption-keys).
+If configured, Coder will use these keys to encrypt external user tokens before
+storing them in the database. The encryption algorithm used is AES-256-GCM with
+a 32-byte key length.
+
+Coder will use the first key provided for both encryption and decryption. If a
+second key is provided, Coder will use it for decryption only. This allows
+administrators to rotate encryption keys without invalidating existing tokens.
+
+The following database fields are currently encrypted:
+
+- `user_links.oauth_access_token`
+- `user_links.oauth_refresh_token`
+- `git_auth_links.oauth_access_token`
+- `git_auth_links.oauth_refresh_token`
+
+Additional database fields may be encrypted in the future.
+
+> Implementation notes: each encrypted database column `$C` has a corresponding
+> `$C_key_id` column. This column is used to determine which encryption key was
+> used to encrypt the data. This allows Coder to rotate encryption keys without
+> invalidating existing tokens, and provides referential integrity for encrypted
+> data.
+>
+> The `$C_key_id` column stores the first 7 bytes of the SHA-256 hash of the
+> encryption key used to encrypt the data.
+>
+> Encryption keys in use are stored in `dbcrypt_keys`. This table stores a
+> record of all encryption keys that have been used to encrypt data. Active keys
+> have a null `revoked_key_id` column, and revoked keys have a non-null
+> `revoked_key_id` column. A key cannot be revoked until all rows referring to
+> it have been re-encrypted with a different key.
```

Comment:
I think this part could just be shortened to "you cannot remove a key until you have rotated all values using that key to a new key"